torch/futures/__init__.py

```diff
@@ -15,7 +33,7 @@ def wait(self):
         """
         return super().wait()
 
-    def then(self, callback):
+    def then(self, callback : Callable[[T], S]) -> torch._C.Future:
```

Comment:
Any reason, this returns `torch._C.Future`, but the API in `torch/_C/__init__.pyi.in` returns a `Future[S]`?

Reply:
Updated, the reason were PEP-0563, but I worked around it by using string typing extension.